Sources/TuistGenerator/Generator/LinkGenerator.swift

```diff
@@ -208,6 +208,8 @@ final class LinkGenerator: LinkGenerating {
     ) throws {
         let embeddableFrameworks = graphTraverser.embeddableFrameworks(path: path, name: target.name).sorted()
 
+        guard !embeddableFrameworks.isEmpty else { return }
```

Comment:
There are cases in which embeddableFrameworks is not empty, but precompiledEmbedPhase ends up being empty (e.g. all embeddableFrameworks are library, bundle, or sdk).
This logic should be plugged in the if branch where we set the script to `"echo \"Skipping, nothing to be embedded.\""`

Reply:
Indeed the distinction is by the nature of the dependency - the results of `embeddableFrameworks` contain a mixture of different dependency types, here are some of the interesting ones:

- `.product()`: A native Xcode target which can be embedded in the "Embed Frameworks"
- `.xcfamework`: External pre-compiled xcframeworks, Xcode knows how to deal with this and can also be added to "Embed Frameworks"
- `.framework`: External pre-compiled `.framework` dependency which goes to the the special script phase _(to strip out the unnecessary slices - this is the methodology that is typically used with Carthage frameworks)_

As @danyf90 only a subset go to the pre-complied phase. 

May be worth extending the test case to cover this?